images/server/install-packages.sh

```diff
@@ -2,26 +2,67 @@
 
 set -ex
 
+
+need_curl() {
+    if command -v curl >/dev/null ; then
+        return 0
+    fi
+    dnf install --setopt=install_weak_deps=False -y /usr/bin/curl
+}
+
 get_custom_repo() {
     url="$1"
     fname="$(basename "$url")"
     dest="/etc/yum.repos.d/${fname}"
-    dnf install --setopt=install_weak_deps=False -y /usr/bin/curl
+    need_curl
     curl -L "$url" -o "$dest"
 }
 
+generate_repo_from_shaman() {
+    url="$1"
+    dest="/etc/yum.repos.d/$2"
+    need_curl
+    tmpfile=/tmp/shaman.json
+    curl -L "$url" -o "${tmpfile}" && python3 <<EOF
+json_file = "${tmpfile}"
+dest = "${dest}"
+import json
+r = json.load(open(json_file))
+url = r[0]["url"]
+ref = r[0]["ref"]
+with open(dest, "w") as out:
+    print(f"[ceph-{ref}]", file=out)
+    print(f"name=Ceph Development Build ({ref})", file=out)
+    print(f"baseurl={url}/x86_64", file=out)
+    print("enabled=1", file=out)
+    print("gpgcheck=0", file=out)
```

Comment:
For me, when I get to the point where I need to embed Python code with shell script, its a sign that I should consider re-writing the entire script in python. It would probably be more maintainable in the long run. (just a though, not a specific action for this PR).

Reply:
Yeah.... it certainly crossed my mind, but the python here is pretty isolated to just extracting the needed data from json. In theory, I could have used jq but that's not already part of the image and python is.